internal/subproviders/morpheus/resources/role/resource.go

```diff
@@ -74,6 +54,454 @@ func (r *Resource) Schema(
+		addRole.SetGlobalSiteAccess(plan.Permissions.DefaultGroupAccess.ValueString())
+	}
+
+	if !plan.Permissions.DefaultInstanceTypeAccess.IsUnknown() {
+		addRole.SetGlobalInstanceTypeAccess(plan.Permissions.DefaultInstanceTypeAccess.ValueString())
+	}
+
+	if !plan.Permissions.DefaultPersonaAccess.IsUnknown() {
+		addRole.SetGlobalPersonaAccess(plan.Permissions.DefaultPersonaAccess.ValueString())
+	}
+
+	if !plan.Permissions.DefaultReportTypeAccess.IsUnknown() {
+		addRole.SetGlobalReportTypeAccess(plan.Permissions.DefaultReportTypeAccess.ValueString())
+	}
+
+	if !plan.Permissions.DefaultTaskAccess.IsUnknown() {
+		addRole.SetGlobalTaskAccess(plan.Permissions.DefaultTaskAccess.ValueString())
+	}
+
+	if !plan.Permissions.DefaultVdiPoolAccess.IsUnknown() {
+		addRole.SetGlobalVdiPoolAccess(plan.Permissions.DefaultVdiPoolAccess.ValueString())
+	}
+
+	if !plan.Permissions.DefaultWorkflowAccess.IsUnknown() {
+		addRole.SetGlobalTaskSetAccess(plan.Permissions.DefaultWorkflowAccess.ValueString())
+	}
+
+	if !plan.Permissions.FeaturePermissions.IsUnknown() {
+		var featurePermissions []FeaturePermissionsValue
+		diags := plan.Permissions.FeaturePermissions.ElementsAs(ctx, &featurePermissions, false)
+		if diags.HasError() {
+			return diags
+		}
+
+		var addRoleFeaturePermissions []sdk.AddRolesRequestRoleFeaturePermissionsInner
+		for _, v := range featurePermissions {
+			addRoleFeaturePermissions = append(addRoleFeaturePermissions, sdk.AddRolesRequestRoleFeaturePermissionsInner{
+				Access: v.Access.ValueString(),
+				Code:   v.Code.ValueString(),
+			})
+		}
+
+		addRole.SetFeaturePermissions(addRoleFeaturePermissions)
+	}
+
+	if !plan.Permissions.BlueprintPermissions.IsUnknown() {
+		var blueprintPermissions []BlueprintPermissionsValue
+		diags = plan.Permissions.BlueprintPermissions.ElementsAs(ctx, &blueprintPermissions, false)
+		if diags.HasError() {
+			return diags
+		}
+
+		var addRoleBlueprintPermissions []sdk.AddRolesRequestRoleAppTemplatePermissionsInner
+		for _, v := range blueprintPermissions {
+			addRoleBlueprintPermissions = append(addRoleBlueprintPermissions, sdk.AddRolesRequestRoleAppTemplatePermissionsInner{
+				Access: v.Access.ValueString(),
+				Id:     int32(v.Id.ValueInt64()),
+			})
+		}
+
+		addRole.SetAppTemplatePermissions(addRoleBlueprintPermissions)
+	}
+
+	if !plan.Permissions.CatalogItemTypePermissions.IsUnknown() {
+		var catalogItemTypePermissions []CatalogItemTypePermissionsValue
+		diags = plan.Permissions.CatalogItemTypePermissions.ElementsAs(ctx, &catalogItemTypePermissions, false)
+		if diags.HasError() {
+			return diags
+		}
+
+		var addRoleCatalogItemTypePermissions []sdk.AddRolesRequestRoleCatalogItemTypePermissionsInner
+		for _, v := range catalogItemTypePermissions {
+			addRoleCatalogItemTypePermissions = append(addRoleCatalogItemTypePermissions, sdk.AddRolesRequestRoleCatalogItemTypePermissionsInner{
+				Access: v.Access.ValueString(),
+				Id:     int32(v.Id.ValueInt64()),
```

Comment:
why cast int64 to int32?

Reply:
It'd probably be a good idea to update this in the openapi spec